samples/sample_plugins/rotate_opencl/src/opencl_filter.cpp

```diff
@@ -127,43 +128,104 @@ cl_int OpenCLFilterBase::InitPlatform()
+                          << " Error code: " << error << endl;
             continue;
+        }
+
+        std::vector<cl_device_id> devices(num_devices);
 
-        if(strstr(platform, "Intel")) // Use only Intel platfroms
+        error = clGetDeviceIDs(*platformIt, CL_DEVICE_TYPE_GPU, devices.size(), &devices[0], 0);
+        if(error)
         {
-            log.info() << "OpenCL platform \"" << platform << "\" is used" << endl;
-            m_clplatform = platforms[platform_index];
+            log.error() << "OpenCLFilter: Couldn't get GPU device IDs."
+                        << " Error code: " << error << endl;
+            continue;
+        }
+
+        for(std::vector<cl_device_id>::iterator deviceIt = devices.begin(); deviceIt != devices.end(); ++deviceIt)
+        {
+            cl_uint deviceVendorId = 0;
+
+            error = clGetDeviceInfo(*deviceIt, CL_DEVICE_VENDOR_ID, sizeof(deviceVendorId), &deviceVendorId, NULL);
+
+            if (error)
+            {
+                log.error() << "OpenCLFilter: Couldn't get the device vendor id."
+                    << " Error code: " << error << endl;
+                continue;
+            }
+
+            // Skip non-Intel devices
+            if (deviceVendorId != 0x8086)
+            {
+                continue;
+            }
+
+            size_t extNamesBufferSize = 0;
+
+            error = clGetDeviceInfo(*deviceIt, CL_DEVICE_EXTENSIONS, 0, NULL, &extNamesBufferSize);
+            if(error)
+            {
+                log.error() << "OpenCLFilter: Couldn't get the size of string with supported extensions for device."
+                            << " Error code: " << error << endl;
+                continue;
+            }
+
+            std::vector<char> extNamesBuffer(extNamesBufferSize);
+
+            error = clGetDeviceInfo(*deviceIt, CL_DEVICE_EXTENSIONS, extNamesBuffer.size(), &extNamesBuffer[0], 0);
+            if(error)
+            {
+                log.error() << "OpenCLFilter: Couldn't get supported extensions for device."
+                            << " Error code: " << error << endl;
+                continue;
+            }
+
+            log.debug() << "OpenCLFilter: Supported extensions for device: " << &extNamesBuffer[0] << endl;
+
+            bool isDeviceAppropriate = true;
+
+            for(std::vector<std::string>::iterator extNameIt = m_requiredOclExtensions.begin(); extNameIt != m_requiredOclExtensions.end(); ++extNameIt)
+            {
+                isDeviceAppropriate = isDeviceAppropriate && (std::strstr(&extNamesBuffer[0], extNameIt->c_str()) != NULL);
+            }
+
+            if(isDeviceAppropriate)
+            {
+                m_clplatform = *platformIt;
+                log.debug() << "OpenCLFilter: Appropriate OCL platform is found." << endl;
+                goto _select_platform_loop_end;
+            }
         }
     }
+
+_select_platform_loop_end:
```

Comment:
you can achieve pretty the same without goto label. Just return from the function when you need to.

Reply:
Changed.